Sources/NIOPosix/SocketChannel.swift

```diff
@@ -526,12 +526,24 @@ final class DatagramChannel: BaseSocketChannel<Socket> {
     }
 
     override func connectSocket(to address: SocketAddress) throws -> Bool {
-        // For now we don't support operating in connected mode for datagram channels.
-        throw ChannelError.operationUnsupported
+        // TODO: this could be a channel option to do other things instead here, e.g. fail the connect
+        if !self.pendingWrites.isEmpty {
```

Comment:
@Lukasa Do I need to consider `self.pendingWrites.isPendingFlush` here? I guess if that's `true` then it means that the API has been used correctly. Or do we still leave it to the user to ensure that the flush has taken place, by means of the associated future?